ads/inabox/position-observer.js

```diff
@@ -142,3 +144,27 @@ function getScrollingElement(win) {
 function isWebKit(ua) {
   return /WebKit/i.test(ua) && !/Edge/i.test(ua);
 }
+
+/**
+ * Get the element's layout rect relative to the viewport.
+ * If parentWin is provided, then attempt to walk up the DOM and add the
+ * offset of all nested parent iframes. Assumes that all parent frames are
+ * friendly and can be inspected (because the element itself can be inspected
+ * as well).
+ * @param {!Element} element
+ * @param {?Window} parentWin The window that contains the element
+ * @return {!LayoutRectDef}
+ */
+export function getTargetRect(element, parentWin) {
+  let targetRect = layoutRectFromDomRect(element./*OK*/getBoundingClientRect());
+  if (parentWin) {
+    for (let j = 0, tempWin = parentWin;
+      j < 10 && tempWin != tempWin.top; j++, tempWin = tempWin.parent) {
```

Comment:
so the assumption is `tempWin.top` is the host window? I feel it's not that reliable. Any better way?

Reply:
I agree. The host window is the window that has the position observer installed, so I moved the function into the class and have the DOM climb stop at that window. The top condition remains just in case.

Reply:
sorry, I still didn't quite get it. where is the code to stop the DOM climb at host window?

Reply:
ah, sorry, was looking at outdated file.